src/map/src/combat/combat.cpp

```diff
@@ -206,17 +206,19 @@ void Combat::update(EntitySystem& entitySystem, Entity entity, uint32_t dt) {
         total_applied_damage += attack.value_;
         adjusted_hp -= attack.value_;
       }
-      
+
+      SrvDamage::DamageAction damage_action = static_cast<SrvDamage::DamageAction>(attack.action_);
+
       if(adjusted_hp <= 0) {
         //TODO: Get dropped item data here and send it with the DAMAGE packet
         attack.action_ &= ~DAMAGE_ACTION_HIT;
         attack.action_ |= DAMAGE_ACTION_DEAD;
-        auto p = SrvDamage::create(attack.attacker_, basicInfo.id, attack.value_, attack.action_);
+        auto p = SrvDamage::create(attack.attacker_, basicInfo.id, attack.value_, damage_action);
```

Comment:
by doing this, nothing will die now. You didn't change `damage_action` to match what it should be.